src/watch.rs

```diff
@@ -95,36 +107,43 @@ where
 
     let (tx_changes, rx_changes) = channel();
 
-    start_event_thread(rx_sources, tx_changes.clone(), ChangeType::Sources);
-    start_event_thread(rx_code, tx_changes, ChangeType::Code);
+    start_event_thread(
+        rx_sources,
+        tx_changes.clone(),
+        ChangeType::Sources,
+        suspend.clone(),
+    );
+    start_event_thread(rx_code, tx_changes, ChangeType::Code, suspend);
 
     Ok((rx_changes, source_watcher, code_watcher))
 }
 
 fn start_event_thread(
-    in_channel: Receiver<DebouncedEvent>,
+    in_channel: Receiver<RawEvent>,
     out_channel: Sender<ChangeType>,
     event_type: ChangeType,
```

Comment:
It is not part of the diff here, but I noticed that you need to `.clone()` this below. Why not `derive(Copy)` on the `enum`?